include/hermes/Platform/Intl/PlatformIntl.h

```diff
@@ -9,6 +9,8 @@
 #define HERMES_PLATFORMINTL_PLATFORMINTL_H
 
 #ifdef HERMES_ENABLE_INTL
+#define UCHAR_TYPE char16_t
```

Comment:
Let's put this in `PlatformIntlICU.cpp` right before the unicode headers with a comment explaining what it does.

I'm slightly surprised that this works fine, but I suppose it is designed to.